### PR TITLE
i186: Fix or remove skipped tests

### LINKED ISSUE #186: Fix or remove skipped tests

### PR DESCRIPTION
**Pull request by nickmeinhold**
fixes #186

### CODE REVIEW

File: pubspec.lock [hunk 1]
```diff
@@ -35,14 +35,14 @@ packages:
       name: async
```
Comment (nickmeinhold) [inline pubspec.lock:35]: I removed the hack for firebase messaging that we added as it's no longer needed

File: pubspec.yaml [hunk 2]
```diff
@@ -6,10 +6,6 @@ version: 0.0.1+1
 environment:
   sdk: ">=2.7.0 <3.0.0"
 
-dependency_overrides:
```
Comment (nickmeinhold) [inline pubspec.yaml:9]: The only change I made was to remove the hack for firebase messaging that we added as it's no longer needed

File: test/mocks/navigation/route_mocks.dart [hunk 3]
```diff
@@ -0,0 +1,253 @@
+import 'package:flutter/material.dart';
```
Comment (nickmeinhold) [inline test/mocks/navigation/route_mocks.dart:1]: I made a stub for the test in `navigation_service_test.dart` - the only functions that do anything but throw UnimplementedError are the first two getters

File: test/unit/services/navigation_service_test.dart [hunk 4]
```diff
@@ -36,16 +35,29 @@ void main() {
     });
 
     test('popHome calls NavigatorState.popUntil with /', () {
-      final navigatorState = MockNavigatorState();
-      final fakeNavKey = FakeGlobalKey(navigatorState);
+      final mockNavigatorState = MockNavigatorState();
+      final fakeNavKey = FakeGlobalKey(mockNavigatorState);
       final service = NavigationService(fakeNavKey);
 
+      /// The service calls [popUntil(ModalRoute.withName('/'))]
```
Comment (nickmeinhold) [inline test/unit/services/navigation_service_test.dart:42]: the test creates the navigation service with a fake nav key (which in turn is created with a "mock navigator state") then calls popHome on the service and verifies the "mock navigator state" was used as expected. 

This is kind of a dumb test and will almost certainly go away when we migrate to Navigator2.0 but it took me frigging ages to get it to work so I can't make myself delete it just yet.

File: pubspec.yaml (hunk 2 above)
Comment (tamari-gray) [reply pubspec.yaml:9]: shweet :)

**Pull request merged**